lib/fog/dynect/models/dns/record.rb

```diff
@@ -59,6 +51,25 @@ def zone
         def zone=(new_zone)
           @zone = new_zone
         end
+
+        def refresh_id
+          begin 
+            records = service.get_record(type, zone.identity, name).body['data']
+          rescue
+            return
+          end
+          # data in format ['/REST/xRecord/domain/fqdn/identity]
+          records.map! do |record|
+            tokens = record.split('/')
+            {
+              :identity => tokens.last,
+              :type     => tokens[2][0...-6] # everything before 'Record'
+            }
+          end
+          record = records.find(ifnone={}) {|record| record[:type] == type}
```

Comment:
I'm not sure you need to say `(ifnone={})`. Perhaps just `({})` would be enough?